etc/testing/travis_build_check_and_stash.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+set -ex
+
+# Travis doesn't reliably give us the right version of the code in PR builds.
+#
+# Sometimes it's too old (the GitHub merge ref API can be out of date) and so
+# we get code from a previous commit in the pull request.
+#
+# Sometimes it's too new (a new commit was pushed to the PR before we ran, and
+# GitHub updated the merge ref API already).
+#
+# We know that we are trying to test code in a merge preview with
+# $TRAVIS_PULL_REQUEST_SHA on the RHS of the merge though, that information is
+# reliable at least.
+#
+# So, try a few times to fetch and identify a valid version of the code. If we
+# time out, give up. If we succeed, stash that code in a docker image which
+# later tests can use to get it back.
+
+cd /home/travis/gopath/src/github.com/pachyderm/pachyderm
```

Comment:
why the cd here?

Reply:
Mostly as documentation of where we are (context for the git commands to operate in) and to make sure we are where we think we are

Reply:
The repo is stored in a directory directly under `/home/travis` though, right? (Maybe `/home/travis/pachyderm`?)

Reply:
It seems to have ended up in the gopath. Not sure how it got there :)

Reply:
Looks like this is magic from `language: go` https://docs.travis-ci.com/user/languages/go/#go-import-path